pkg/controllers/mcs/service_export_controller_test.go

```diff
@@ -0,0 +1,1978 @@
+	_ = clusterv1alpha1.AddToScheme(scheme)
+	return scheme
+}
+
+// Mock structures for testing
+
+// mockErrorClient wraps a client and allows configuring errors for specific operations
+type mockErrorClient struct {
+	client.Client
+	getError    error
+	listError   error
+	deleteError error
+	updateError error
+}
+
+func (e *mockErrorClient) Get(_ context.Context, _ types.NamespacedName, _ client.Object, _ ...client.GetOption) error {
+	if e.getError != nil {
+		return e.getError
+	}
+	return e.Client.Get(context.TODO(), types.NamespacedName{}, nil)
+}
+
+func (e *mockErrorClient) List(_ context.Context, _ client.ObjectList, _ ...client.ListOption) error {
+	if e.listError != nil {
+		return e.listError
+	}
+	return e.Client.List(context.TODO(), nil)
+}
+
+func (e *mockErrorClient) Delete(_ context.Context, _ client.Object, _ ...client.DeleteOption) error {
+	if e.deleteError != nil {
+		return e.deleteError
+	}
+	return e.Client.Delete(context.TODO(), nil)
+}
+
+func (e *mockErrorClient) Update(_ context.Context, _ client.Object, _ ...client.UpdateOption) error {
+	if e.updateError != nil {
+		return e.updateError
+	}
+	return e.Client.Update(context.TODO(), nil)
+}
+
+// mockLister simulates cache.GenericLister
+type mockLister struct {
+	getFunc  func(name string) (runtime.Object, error)
+	listFunc func(selector labels.Selector) ([]runtime.Object, error)
+}
+
+func (m *mockLister) List(selector labels.Selector) ([]runtime.Object, error) {
+	if m.listFunc != nil {
+		return m.listFunc(selector)
+	}
+	return nil, nil
+}
+
+func (m *mockLister) Get(name string) (runtime.Object, error) {
+	if m.getFunc != nil {
+		return m.getFunc(name)
+	}
+	return nil, apierrors.NewNotFound(schema.GroupResource{}, name)
+}
+
+func (m *mockLister) ByNamespace(_ string) cache.GenericNamespaceLister {
+	return &mockNamespaceLister{lister: m}
+}
+
+// mockNamespaceLister simulates cache.GenericNamespaceLister
+type mockNamespaceLister struct {
+	lister *mockLister
+}
+
+func (m *mockNamespaceLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	return m.lister.List(selector)
+}
+
+func (m *mockNamespaceLister) Get(name string) (runtime.Object, error) {
+	return m.lister.Get(name)
+}
+
+func TestCleanEndpointSliceWork(t *testing.T) {
+	ctx := context.TODO()
+
+	tests := []struct {
+		name    string
+		prepare func() (client.Client, *workv1alpha1.Work)
+		verify  func(t *testing.T, c client.Client, originalWork *workv1alpha1.Work, err error)
+	}{
+		{
+			name: "should delete work when only managed by ServiceExport",
+			prepare: func() (client.Client, *workv1alpha1.Work) {
+				work := &workv1alpha1.Work{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-work",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							util.EndpointSliceWorkManagedByLabel: util.ServiceExportKind,
+						},
+					},
+				}
+				client := fake.NewClientBuilder().WithScheme(newTestScheme()).WithObjects(work).Build()
+				return client, work
```

Comment:
Variable 'client' collides with imported package name.
```suggestion
				
				return fake.NewClientBuilder().WithScheme(newTestScheme()).WithObjects(work).Build(), work
```
Same as below.